### DISCUSSION TITLE
How to turn off / stop / set to stand by the GNSS

### DISCUSSION THREAD

**Discussion by JosanV** [semuconsulting/pyubx2#154] [Q&A] [answered]
Hi all, I am using the pyubx2 library to communicate with an EVK-M9DR device and I would like to know if it is possible to put the GNSS in standby mode or stop it directly from the library. Of course, I would also be interested in knowing how to turn it back on.

I've been going through the documentation and available configuration options, such as power management (PM) keys and other hardware settings, but I'm not sure what would be the right way to do it. 
I have read a post, but the library has changed since it was created:
https://github.com/semuconsulting/pyubx2/issues/11

From u-center it can be done from UBX - CFG - PWR or also from UBX - CFG - RST, but I can't find the command in:
https://github.com/semuconsulting/pyubx2/blob/master/src/pyubx2/ubxtypes_configdb.py

If anyone has had experience with this or knows how this can be accomplished, could you share any examples or provide guidance on how to configure the device to go into standby mode or stop GNSS?

Thanks in advance.

**Comment by semuadmin** [comment, marked as answer]
Hi @JosanV,

You can use the UBX CFG-RST and/or CFG-PWR commands to stop or start the GNSS module in your device, e.g.

```python
from pyubx2 import UBXMessage, SET

# power-down GNSS using CFG-PWR
msg = UBXMessage("CFG", "CFG-PWR", SET, state=0x53544F50)
print(msg)

# restart GNSS using CFG-RST
msg = UBXMessage("CFG", "CFG-RST", SET, resetMode=0x09)
print(msg)
```

Refer to your device's [Interface Specification](https://www.u-blox.com/en/product/evk-m9dr?legacy=Current#Documentation-&-resources) for further details and the appropriate values for the `state` and `resetMode` attributes for each of the various reset or power modes e.g.

<img width="812" alt="Screenshot 2024-06-10 at 12 31 14" src="https://github.com/semuconsulting/pyubx2/assets/28569967/ceebd618-75c1-4c7d-8d64-fccbcd45cfc0">

You could add these commands as [User Presets](https://github.com/semuconsulting/PyGPSClient?tab=readme-ov-file#userdefined) in PyGPSClient if you envisage using them frequently.

**Reply by JosanV** [reply]
That worked perfectly as expected! Thank you @semuadmin

```
msg = UBXMessage("CFG", "CFG-RST", SET, resetMode=0x08) # STOP
msg = UBXMessage("CFG", "CFG-RST", SET, resetMode=0x09) # START
send_message(stream, serial_lock, msg)
```
